README.md

```diff
@@ -0,0 +1,399 @@
+# `turludock` - Robot Operating System (ROS) Docker Containers with X11 and Wayland support for Linux
+[![N|Solid](http://turlucode.com/wp-content/uploads/2017/10/turlucode_.png)](http://turlucode.com/)
+
+`turludock` aims to provide different versions of ROS as docker containers, but with GUI 
+support! This means your local OS is no more bound to the version of ROS you are using! 
+You can use _any_ version of ROS with _any_ Linux distribution, thanks to the amazing 
+power of docker!
+
+# TL;DR
+## Build image
+```sh
+# Install tool
+pip install turludock
+
+# Build ROS image from presets (turludock which presets)
+turludock build -e noetic_mesa
+# Build from manual config 
+turludock build -c custom.yaml --tag custom_tag
+
+# Check supported versions
+turludock which ros
+turludock which cuda ROS_CODENAME
+```
+All commands have a `--help` support.
+> [Example YAML config](examples/noetic_nvidia_custom.yaml)
+
+## Run container
+[See here](#running-the-image-as-current-user) how to run a container.
+
+# Getting Started
+The idea is to have HW accelerated GUIs with docker. Generally it has proven that this is 
+a challenging task. Graphics card companies like NVIDIA, already provide 
+solutions for their platform. Running X11 applications within a docker container is proven for
+the last years. However, Wayland support remains still a bit challenging, but at least for
+our ROS containers we can assume it is supported :cake:.
+
+## Install `turludock`
+```sh
+pip install turludock
+```
+
+## Supported tool functionality
+This tool has basically two main functionalities:
```

Comment:
I would say:  "This tool has two functionalities:"
otherwise the reader might ask, what are the other functionalities?